PdhLookupPerfIndexByEnglishName causes ExceptionInInitializerError

Exception in thread "GetHWID Thread" java.lang.ExceptionInInitializerError
        at oshi.hardware.platform.windows.WindowsHardwareAbstractionLayer.getDiskStores(Unknown Source)
        at ru.gravit.launcher.s6cWIgQFaKFozpiF0rvCUkV4.tIxukqwJpjbqRiR8fRRJ9oyw(
Unknown Source)
        at ru.gravit.launcher.s6cWIgQFaKFozpiF0rvCUkV4.OvndpPHp1rqtyiydNxBfVvo1(
Unknown Source)
        at ru.gravit.launcher.DJBo2tP0OoFaOZANQ00ZcxtJ.getHWID(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: com.sun.jna.platform.win32.Win32Exception: Не удается найти указанный
 файл.
        at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Unknow
n Source)
        at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Unknow
n Source)
        at com.sun.jna.platform.win32.Advapi32Util.registryGetStringArray(Unknow
n Source)
        at com.sun.jna.platform.win32.PdhUtil.PdhLookupPerfIndexByEnglishName(Un
known Source)
        at oshi.hardware.platform.windows.WindowsDisks.<clinit>(Unknown Source)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,6 @@
-4.0.0 (in progress)
+3.12.3 (in progress)
 ================
-* Your contribution here.
+* [#764](https://github.com/oshi/oshi/pull/764): Prevent exception in WindowsDisks initializer. - [@dbwiddis](https://github.com/dbwiddis).
 
 3.12.1 (12/31/2018), 3.12.2 (1/10/2019)
 ================

--- a/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsDisks.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsDisks.java
@@ -39,6 +39,7 @@ import com.sun.jna.platform.win32.Kernel32; // NOSONAR squid:S1191
 import com.sun.jna.platform.win32.PdhUtil;
 import com.sun.jna.platform.win32.PdhUtil.PdhEnumObjectItems;
 import com.sun.jna.platform.win32.PdhUtil.PdhException;
+import com.sun.jna.platform.win32.Win32Exception;
 import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiQuery;
 import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiResult;
 
@@ -129,10 +130,11 @@ public class WindowsDisks implements Disks {
     private static WmiQuery<PhysicalDiskProperty> physicalDiskQuery = null;
 
     static {
-        String physicalDisk = PdhUtilXP.PdhLookupPerfNameByIndex(null,
-                PdhUtil.PdhLookupPerfIndexByEnglishName(PHYSICAL_DISK));
+        String physicalDisk = PHYSICAL_DISK;
         boolean enumeration = true;
         try {
+            physicalDisk = PdhUtilXP.PdhLookupPerfNameByIndex(null,
+                    PdhUtil.PdhLookupPerfIndexByEnglishName(PHYSICAL_DISK));
             PdhEnumObjectItems objectItems = PdhUtil.PdhEnumObjectItems(null, null, physicalDisk, 100);
             if (!objectItems.getInstances().isEmpty()) {
                 List<String> instances = objectItems.getInstances();
@@ -176,7 +178,7 @@ public class WindowsDisks implements Disks {
                     }
                 }
             }
-        } catch (PdhException e) {
+        } catch (PdhException | Win32Exception e) {
             LOG.warn("Unable to enumerate performance counter instances for {}.", physicalDisk);
             enumeration = false;
         }

